src/com/google/javascript/jscomp/ProcessCommonJSModules.java

```diff
@@ -140,6 +188,25 @@ public void visit(NodeTraversal t, Node n, Node parent) {
         visitRequireCall(t, n, parent);
       }
 
+      // Check for UMD pattern, pull out CommonJS branch and remove the rest.
```

Comment:
Could you add some more documentation on what this exactly does? "pull out" is a bit unclear. Maybe give a succinct example of what pattern it detects and rewrites?